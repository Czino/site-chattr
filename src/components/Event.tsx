import { NDKEvent } from '@nostr-dev-kit/ndk'
import { useIsSelf } from '../hooks/useIsSelf'

const DEFAULTS = {
    USER_NAME: 'Anon',
}
const MS_PER_S = 1000
const GATEWAY = 'https://njump.me/'
export const Event = ({ event }: { event: NDKEvent }) => {
    const isSelf = useIsSelf()

    return (
        <li
            className={[
                'w-full rounded p-4 overflow-hidden flex flex-col gap-1',
                isSelf(event.author.pubkey) ? 'bg-highlight' : 'bg-purple-100',
            ].join(' ')}
        >
            <a className="flex gap-1 items-center leading-5"
                href={GATEWAY + event.author.npub}
                target="_blank"
                rel="noreferrer"
            >
                {event.author.profile?.image ? (
                    <img src={event.author.profile?.image} alt="" className="w-5 h-5 rounded-full" />
                ) : (
                    <div className="w-5 h-5 rounded-full bg-purple-400"></div>
                )}
                <p className="font-bold">{event.author.profile?.displayName || DEFAULTS.USER_NAME}</p>
                {!!event.created_at && (
                    <p className="text-purple-400 text-xs">
                        {new Date(event.created_at * MS_PER_S).toLocaleString()}
                    </p>
                )}
            </a>
            <p>{event.content}</p>
        </li>
    )
}
