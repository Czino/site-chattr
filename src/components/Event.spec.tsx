import NDK, { NDKEvent } from '@nostr-dev-kit/ndk'
import { render } from '@testing-library/react'
import { event1, event2 } from '../../test/data/eventData'
import { Event } from './Event'

const pubkey = 'c446c97e0b3a2f5e1d33ba40664fa77b03b0afcba9684daf4f7925ae8f5c6fe0'
const ndk = new NDK()
const event = new NDKEvent(ndk, event1)
jest.mock('nostr-hooks', () => ({
    useActiveUser: () => ({
        activeUser: { pubkey },
    }),
}))
describe('Event', () => {
    const base = render(<Event event={event} />).asFragment()
    it('should render correctly', () => {
        expect(base).toMatchSnapshot()
    })
    it('should render profile image', () => {
        event.author.profile = { image: 'url/to/image.png' }
        const { asFragment } = render(<Event event={event} />)
        expect(asFragment()).toMatchDiffSnapshot(base)
    })
    it('should highlight event if note is from active user', () => {
        const otherEvent = new NDKEvent(ndk, event2)

        const { asFragment } = render(<Event event={otherEvent} />)
        expect(asFragment()).toMatchDiffSnapshot(base)
    })
})
