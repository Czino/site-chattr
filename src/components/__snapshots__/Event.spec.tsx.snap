// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event should highlight event if note is from active user 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -1,12 +1,12 @@
  <DocumentFragment>
    <li
-     class=\\"w-full rounded p-4 overflow-hidden flex flex-col gap-1 bg-highlight\\"
+     class=\\"w-full rounded p-4 overflow-hidden flex flex-col gap-1 bg-purple-100\\"
    >
      <a
        class=\\"flex gap-1 items-center leading-5\\"
-       href=\\"https://njump.me/npub1c3rvjlst8gh4u8fnhfqxvna80vpmpt7t495ymt600yj6ar6udlsq60swvh\\"
+       href=\\"https://njump.me/npub1xunyx3crpkgem3nntqudrrg5pnp329q27p5yz59ggccc7lq604vsan554l\\"
        rel=\\"noreferrer\\"
        target=\\"_blank\\"
      >
        <div
          class=\\"w-5 h-5 bg-purple-400 rounded-full\\"
@@ -17,13 +17,13 @@
          Anon
        </p>
        <p
          class=\\"text-xs text-purple-400\\"
        >
-         9/18/2024, 10:04:39 AM
+         9/17/2024, 2:47:01 PM
        </p>
      </a>
      <p>
-       I develop almost all my apps here!
+       Home sweet home
      </p>
    </li>
  </DocumentFragment>"
`;

exports[`Event should render correctly 1`] = `
<DocumentFragment>
  <li
    class="w-full rounded p-4 overflow-hidden flex flex-col gap-1 bg-purple-100"
  >
    <a
      class="flex gap-1 items-center leading-5"
      href="https://njump.me/npub1xunyx3crpkgem3nntqudrrg5pnp329q27p5yz59ggccc7lq604vsan554l"
      rel="noreferrer"
      target="_blank"
    >
      <div
        class="w-5 h-5 bg-purple-400 rounded-full"
      />
      <p
        class="font-bold"
      >
        Anon
      </p>
      <p
        class="text-xs text-purple-400"
      >
        9/17/2024, 2:47:01 PM
      </p>
    </a>
    <p>
      Home sweet home
    </p>
  </li>
</DocumentFragment>
`;

exports[`Event should render profile image 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -6,14 +6,12 @@
        class=\\"flex gap-1 items-center leading-5\\"
        href=\\"https://njump.me/npub1xunyx3crpkgem3nntqudrrg5pnp329q27p5yz59ggccc7lq604vsan554l\\"
        rel=\\"noreferrer\\"
        target=\\"_blank\\"
      >
-       <img
-         alt=\\"\\"
-         class=\\"w-5 h-5 rounded-full\\"
-         src=\\"url/to/image.png\\"
+       <div
+         class=\\"w-5 h-5 bg-purple-400 rounded-full\\"
        />
        <p
          class=\\"font-bold\\"
        >
          Anon"
`;
